### PR TITLE
Restructure gem to conform to conventional directory structure (based…

### PR DESCRIPTION
**Pull request by olbrich**
… on what bundle gem produces)

The original directory structure of this gem does not conform to the recommended structure for a ruby gem.

1. Ran `bundle gem . --test=test-unit` and resolved conflicts in files that already existed.
2. moved most files under `lib/fhir_client`
3. fixed `fhir_client.rb` to load them properly
4. removed a couple of gems that don't seem to be used, or which were deprecated (i.e, turn, minitest)
5. adjusted the rubocop settings a bit since the generated one seems to ignore almost all rules
6. renamed some camelCased methods and variables to the under_scored version
7. I required some of the dependency gems in the files where they are used to make it easier to see where things are needed.

This also provides the standard bundler gem tasks for publishing/packaging gems, a bin/console command.

Note that the version number is now controlled by the `lib/fhir_client/version` file.

### CODE REVIEW

File: lib/fhir_client.rb [hunk 1]
```diff
@@ -1,26 +1,23 @@
-# Top level include file that brings in all the necessary code
-require 'bundler/setup'
-require 'yaml'
-require 'nokogiri'
+require 'fhir_client/version'
+
+# require 'yaml'  # not sure what (if anything) this is needed for
```
Comment (olbrich) [inline lib/fhir_client.rb]: I was unable to determine why this was required and removing it does not cause any tests to fail.

File: lib/fhir_client/client.rb [hunk 2]
```diff
@@ -1,3 +1,8 @@
+require 'rest_client'
```
Comment (olbrich) [inline lib/fhir_client/client.rb]: ideally files should be named after the class they contain.

File: test/test_helper.rb [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
 
 require 'simplecov'
-require_relative '../lib/fhir_client'
-
 require 'pry'
-require 'minitest/autorun'
-require 'webmock/test_unit'
-require 'bundler/setup'
 require 'test/unit'
-require 'turn'
```
Comment (olbrich) [inline test/test_helper.rb]: this gem is deprecated

File: lib/fhir_client.rb (hunk 1 above)
Comment (jawalonoski) [inline lib/fhir_client.rb]: See the `request_patch_payload` method in `client_interface.rb`

Comment (olbrich) [inline lib/fhir_client.rb]: How does that method depend on 'yaml' ?

Comment (jawalonoski) [inline lib/fhir_client.rb]: I was referencing nokogiri, not yaml.

Comment (olbrich) [inline lib/fhir_client.rb]: ah ok.  Nokogiri is still there, it's just been moved.

My original comment was about the 'yaml' dependency.

**Pull request merged**